modules/RTC/TPCUtils.js

```diff
@@ -485,43 +504,39 @@ export class TPCUtils {
             return Promise.resolve();
         }
         parameters.encodings = this._getStreamEncodings(track);
-        const promise = transceiver.sender.setParameters(parameters);
 
         if (mediaType === MediaType.VIDEO) {
-            return this.pc._updateVideoSenderParameters(promise);
+            return this.pc._updateVideoSenderParameters(() => transceiver.sender.setParameters(parameters));
         }
 
-        return promise;
+        return transceiver.sender.setParameters(parameters);
     }
 
     /**
      * Resumes or suspends media on the peerconnection by setting the active state on RTCRtpEncodingParameters
      * associated with all the senders that have a track attached to it.
      *
-     * @param {boolean} enable - whether media needs to be enabled or suspended.
+     * @param {boolean} enable - whether outgoing media needs to be enabled or disabled.
+     * @param {string} mediaType - media type, 'audio' or 'video', if neither is passed, all outgoing media will either
+     * be enabled or disabled.
      * @returns {Promise} - A promise that is resolved when the change is succesful on all the senders, rejected
      * otherwise.
      */
-    setMediaTransferActive(enable) {
+    setMediaTransferActive(enable, mediaType) {
         logger.info(`${this.pc} ${enable ? 'Resuming' : 'Suspending'} media transfer.`);
 
-        const senders = this.pc.peerconnection.getSenders().filter(s => Boolean(s.track));
+        const senders = this.pc.peerconnection.getSenders()
+            .filter(s => Boolean(s.track) && (!mediaType || s.track.kind === mediaType));
         const promises = [];
 
         for (const sender of senders) {
-            const parameters = sender.getParameters();
-
-            if (parameters?.encodings?.length) {
-                for (const encoding of parameters.encodings) {
-                    encoding.active = enable;
-                }
-            }
-            const setActivePromise = sender.setParameters(parameters);
-
             if (sender.track.kind === MediaType.VIDEO) {
-                promises.push(this.pc._updateVideoSenderParameters(setActivePromise));
+                promises.push(this.pc._updateVideoSenderParameters(this._updateSenderParameters(sender, enable)));
             } else {
-                promises.push(setActivePromise);
+                const parameters = sender.getParameters();
+
+                parameters?.encodings?.length && (parameters.encodings[0].active = enable);
```

Comment:
Why we don't use `this._updateSenderParameters` here?

Reply:
Yes we can, I will update it.